### PR TITLE
Fix SelectField props validation

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
### Summary

During the [update of the MC-frontend](https://github.com/commercetools/merchant-center-frontend/pull/11847) to the newest version of ui-kit (v13) there have been found some errors from the applications using the `SelectField` component when `isMulti` property is _true_.

### Description

The problem relies in that the component perform some runtime validations on properties `value` and `touched` for them to be arrays in case `isMulti` is _true_.

Both those properties are declares as optional, but when using `isMulti` _true_ we're expecting (incorrectly) for them to exist.

### CODE REVIEW

File: packages/components/fields/select-field/src/select-field.tsx [hunk 1]
```diff
@@ -308,12 +308,14 @@ export default class SelectField extends Component<TSelectFieldProps> {
 
     if (this.props.isMulti) {
       warning(
-        Array.isArray(this.props.value),
+        Array.isArray(this.props.value) ||
+          typeof this.props.value === 'undefined',
         'SelectField: `value` is expected to be an array of string when isMulti is true'
```
Comment (emmenko) [inline packages/components/fields/select-field/src/select-field.tsx:313]:
```suggestion
        `SelectField: "value" is expected to be an array of string when "isMulti" is true, instead got ${this.props.value}.`
```

File: packages/components/fields/select-field/src/select-field.tsx [hunk 2]
```diff
@@ -308,12 +308,14 @@ export default class SelectField extends Component<TSelectFieldProps> {
...
       );
 
       warning(
-        Array.isArray(this.props.touched),
+        Array.isArray(this.props.touched) ||
+          typeof this.props.touched === 'undefined',
         'SelectField: `touched` is expected to be an array of boolean when isMulti is true'
```
Comment (emmenko) [inline packages/components/fields/select-field/src/select-field.tsx:319]:
```suggestion
        `SelectField: "touched" is expected to be an array of boolean when "isMulti" is true, instead got ${this.props.touched}.`
```

File: .changeset/slow-cameras-brush.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-uikit/select-field': patch
```
Comment (emmenko) [inline .changeset/slow-cameras-brush.md:2]:
```suggestion
'@commercetools-uikit/select-field': patch
'@commercetools-uikit/select-input': patch
```

File: packages/components/fields/select-field/src/select-field.tsx (hunk 1 above)
Comment (CarlosCortizasCT) [reply packages/components/fields/select-field/src/select-field.tsx]: Updated in 60d2ee841c61658c25251cd1b69ce397c929b0de

Thanks!

File: packages/components/fields/select-field/src/select-field.tsx (hunk 2 above)
Comment (CarlosCortizasCT) [reply packages/components/fields/select-field/src/select-field.tsx]: Updated in 60d2ee841c61658c25251cd1b69ce397c929b0de

Thanks!

File: .changeset/slow-cameras-brush.md (hunk 3 above)
Comment (CarlosCortizasCT) [reply .changeset/slow-cameras-brush.md:2]: Added another changeset file for `select-input` so we have different messages per package: 6ad608a0c3884dec23057a35bda03005c368b6c9

Thanks~

**Review by emmenko — APPROVED** [review]
💯

**Pull request merged**